dotCMS/src/main/java/com/liferay/portal/servlet/MainServlet.java

```diff
@@ -106,12 +106,11 @@ public void init(ServletConfig config) throws ServletException {
 
 			// Checking for execute upgrades
 			try {
-				StartupTasksExecutor.getInstance().executeUpgrades(config.getServletContext().getRealPath("/"));
-			} catch (DotRuntimeException e1) {
+				StartupTasksExecutor.getInstance().executeStartup();
+				StartupTasksExecutor.getInstance().executeUpgrades();
+			} catch (Exception e1) {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)